source/CardGames.py

```diff
@@ -238,7 +239,18 @@ def initializeGame():
     anotherPlayer = input("Add another player? (Y/N): ")
     if anotherPlayer == "Y" or anotherPlayer == "y":
       newPlayer = input("Enter name of new player: ")
-      playerNames.append(newPlayer)
+      while checkDuplicate(newPlayer) == True:
+        useExisting = input(f"Player name already exists. Continue as {newPlayer}? (Y/N): ")
+        if useExisting == "Y" or useExisting == "y":
+          playerNames.append(newPlayer)
+          break
+        else:
+          newName = input("Please enter new player name: ")
+          while checkDuplicate(newName) == True:
+            newName = input("Player name already exists. Please enter new name: ")
+            checkDuplicate(newName)
+          playerNames.append(newName)
+          break
```

Comment:
I think this is a good implementation for not allowing duplicates but I don't see anywhere where you're referencing the leaderboard file to check for duplicates there.